service/frontend/adminHandler.go

```diff
@@ -595,63 +605,75 @@ func (adh *AdminHandler) updateEventRange(
 	if err != nil {
 		return nil, err
 	}
+	firstItem, err := targetBranch.GetFirstItem()
+	if err != nil {
+		return nil, err
+	}
+	lastItem, err := targetBranch.GetLastItem()
+	if err != nil {
+		return nil, err
+	}
+
 	if request.StartEventId == nil || request.StartEventVersion == nil {
-		firstItem, err := targetBranch.GetFirstItem()
-		if err != nil {
-			return nil, err
-		}
 		// If start event is not set, get the events from the first event
 		// As the API is exclusive-exclusive, use first event id - 1 here
 		request.StartEventId = common.Int64Ptr(common.FirstEventID - 1)
 		request.StartEventVersion = common.Int64Ptr(firstItem.GetVersion())
 	}
-	isEndBoundarySet := true
 	if request.EndEventId == nil || request.EndEventVersion == nil {
-		lastItem, err := targetBranch.GetLastItem()
-		if err != nil {
-			return nil, err
-		}
 		// If end event is not set, get the events until the end event
-		// As the API is exclusive-exclusive, use end event id + 1 1 here
+		// As the API is exclusive-exclusive, use end event id + 1 here
 		request.EndEventId = common.Int64Ptr(lastItem.GetEventID() + 1)
 		request.EndEventVersion = common.Int64Ptr(lastItem.GetVersion())
-		isEndBoundarySet = false
 	}
 
 	if request.GetStartEventId() < 0 {
 		return nil, &gen.BadRequestError{Message: "Invalid FirstEventID && NextEventID combination."}
 	}
 
-	// get branch based on the end event
-	if isEndBoundarySet {
+	// get branch based on the end event if end event is defined in the request
+	if request.GetEndEventId() != lastItem.GetEventID()+1 &&
```

Comment:
I would do the follow
```go
if request.GetEndEventId() == lastItem.GetEventID()+1 && request.GetEndEventVersion() != lastItem.GetVersion() {
    // this is a special case, target branch remains the same
} else {
    // this is common case
    <put the code block here>
}

Reply:
I would do the follow

if request.GetEndEventId() == lastItem.GetEventID()+1 && request.GetEndEventVersion() != lastItem.GetVersion() {
    // this is a special case, target branch remains the same
} else {
    // this is common case
    <put the code block here>
}